src/Core/Commit.js

```diff
@@ -0,0 +1,314 @@
+ * @param {number} signature
+ */
+function commitUnmount (element, parent, signature) {
+	if (element.flag === ElementComponent)
+		return componentUnmount(element, element.children, parent, signature, 1)
+
+	commitRemove(element, parent)
+	commitDetach(element, signature)
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} snapshot
+ * @param {number} signature
+ */
+function commitReplace (element, snapshot, signature) {
+	if (signature > 0 && commitUnmount(element, element.parent, 1))
+		return void element.state.then(function () {
+			commitReplace(element, snapshot, 0)
+		})
+
+	commitMount(snapshot, elementSibling(element, 0), element.parent, element.host, 1)
+
+	for (var key in snapshot)
+		switch (key) {
+			case 'DOM':
+				merge(element[key], snapshot[key])
+			case 'next':
+			case 'prev':
+				break
+			default:
+				element[key] = snapshot[key]	
+		}
+}
+
+/**
+ * @param {Element} element
+ * @param {number} signature
+ */
+function commitDetach (element, signature) {
+	if (element.flag !== ElementText) {
+		var index = 0
+		var children = element.children
+		var length = children.length
+		var next = children.next
+
+		while (index++ < length)
+			switch (next.flag) {
+				case ElementComponent:
+					if (next.owner[LifecycleWillUnmount])
+						lifecycleMount(next, LifecycleWillUnmount)
+				default:
+					commitDetach(next, 1)
+					next = next.next
+			}
+	}
+
+	if (element.ref)
+		commitRef(element, element.ref, -1)
+
+	if (signature < 1) {
+		element.context = null
+		element.state = null
+		element.event = null
+		element.DOM = null
+	}
+}
+
+/**
+ * @param {Element} element
+ * @param {(function|string)?} callback
+ * @param {number} signature
+ * @param {*} key
+ */
+function commitRef (element, callback, signature, key) {
+	switch (typeof callback) {
+		case 'string':
+			return commitRef(element, componentRef, signature, callback)			
+		case 'function':
+			switch (signature) {
+				case -1:
+					return lifecycleCallback(element.host, callback, element.ref = null, key, element)
+				case 0:
+					element.ref = callback
+				case 1:
+					return lifecycleCallback(element.host, callback, element.instance || findDOMNode(element), key, element)
+				case 2:
+					commitRef(element, callback, -1, key)
+					commitRef(element, callback, 0, key)
+			}
+			break
+		default:
+			commitRef(element, element.ref || noop, -1, key)
+	}
+}
+
+/**
+ * @param {Element} element
+ * @param {string} type
+ * @param {(function|EventListener)} callback
+ */
+function commitEvent (element, type, callback) {
+	if (!element.event)
+		element.event = {}
+	
+	element.event[type] = callback
+	Event.addEventListener(element, type)
+}
+
+/**
+ * @param {Element} element
+ * @param {Object} props
+ * @param {number} signature
+ */
+function commitProps (element, props, signature) {
+	for (var key in props)
+		if (key === 'ref')
+			commitRef(element, props[key], signature)
+		else if (key.charCodeAt(0) === 111 && key.charCodeAt(1) === 110)
```

Comment:
add a comment with char code mappings - `111 == /` `110 == .`
so props of `./xxx` are event listeners?

Reply:
Sure, props of `on...`.